src/foam/dao/VersionJDAO.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.ClassInfo;
+import foam.lib.json.Outputter;
+import foam.lib.json.OutputterMode;
+import java.io.*;
+
+public class VersionJDAO
```

Comment:
I don't think we want to do this as a sub-class because if we then want a HashingJDAO or a SigningJDAO or whatever to have this feature, we'll then need a VersionHashingJDAO and a VersionSigningJDAO also. This will lead to an exponential explosion of JDAO sub-classes.

Reply:
I would rather build this into the base AbstractJDAO functionality so that all JDAO's can use this functionality.
Maybe, just have it look for filename.0 and load it if available, then use filename, just like normal. This way it works just like normal if the filename.0 or filename.base file doesn't exist.